cmd/proxy/Dockerfile

```diff
@@ -1,15 +1,30 @@
-FROM golang:1.11
+FROM golang:1.11-alpine AS builder
 
-RUN mkdir -p $GOPATH/src/github.com/gomods/athens
 WORKDIR $GOPATH/src/github.com/gomods/athens
 
 COPY . .
 
-RUN CGO_ENABLED=0 GO111MODULE=on go build -mod=vendor -o /bin/app ./cmd/proxy
-RUN ./scripts/create_default_config.sh
+RUN apk update && \ 
+	apk add --no-cache bash && \
+	GO111MODULE=on CGO_ENABLED=0 go build -mod=vendor -o /bin/app ./cmd/proxy && \
+	./scripts/create_default_config.sh
+
+FROM alpine
+
+ENV GO111MODULE=on
+
+COPY --from=builder /bin/app /bin/app
+COPY --from=builder /go/src/github.com/gomods/athens/config.toml /go/src/github.com/gomods/athens/config.toml
+COPY --from=builder /go/src/github.com/gomods/athens/cmd/proxy/locales /go/src/github.com/gomods/athens/cmd/proxy/locales
+COPY --from=builder /go/src/github.com/gomods/athens/cmd/proxy/assets /go/src/github.com/gomods/athens/cmd/proxy/assets
+COPY --from=builder /usr/local/go/bin/go /bin/go
+
+RUN apk update &&  \
+		apk add --no-cache bzr git mercurial openssh-client subversion procps && \
```

Comment:
I wouldn't know, believe whomever does the project testing would know this information.  This is simply the line from the original image that was being used and supplied the required VCS applications.

Reply:
@gcstang that's good to know, can you put a link to that exact line? So we can always compare in case Go introduces a new VCS dependency or changes something. Thank you!

Reply:
It's part of the buildpacks-stretchscm release that golang uses in this file: https://github.com/docker-library/buildpack-deps/blob/1845b3f918f69b4c97912b0d4d68a5658458e84f/stretch/scm/Dockerfile

Reply:
https://github.com/golang/go/blob/f082dbfd4f23b0c95ee1de5c2b091dad2ff6d930/src/cmd/go/internal/get/vcs.go#L90

Reply:
I'm not familiar with anything other than Git, but can we: 
1. double check that the array @tnextday mentioned is all installed in this dockerfile? 
2. Add either/both links to the Dockerfile itself so it's easy in the future to compare if things get outdated (which they will) 

Thank you!

Reply:
Also need to add the [fossil](https://pkgs.alpinelinux.org/packages?name=fossil&branch=v3.8) package.

Reply:
added